src/phoenix/server/api/mutations/chat_mutations.py

```diff
@@ -174,6 +175,14 @@ async def chat_completion_over_dataset(
                     )
                 )
             ]
+
+            username: Optional[str] = None
+            if user := info.context.user and get_env_enable_auth():
+                user_id = user.identity
+                db_user = await session.scalar(select(models.User).filter_by(id=int(user_id)))
+                if db_user:
+                    username = db_user.username
```

Comment:
You could be using a data loader here I believe. Also isn't the DBuser on the context?

Reply:
I would do that. Otherwise this is inefficient

Reply:
the username isn't, just the id